src/nvim/spell.c

```diff
@@ -3648,7 +3645,8 @@ static void suggest_trie_walk(suginfo_T *su, langp_T *lp, char_u *fword, bool so
   bool goodword_ends;
 #ifdef DEBUG_TRIEWALK
   // Stores the name of the change made at each level.
-  char_u changename[MAXWLEN][80];
+  static const size_t changenameElemSize = 80;
```

Comment:
What is the advantage of the additional `static` here?

Reply:
It narrowa the scope of the variable to current file.

Reply:
It narrows it down to function scope and keeps the value between function calls. But why do we want this instead of just a constant value? Is it more efficient?

Reply:
@ZviRackover It does not, not in this context. Neither it does “narrow down to function scope”, all variables declared inside function already have only this scope, no matter `static` or not. Just `static` variables declared in a function are essentially *global* variables; you would not get a symbol for them so you can’t just access that outside of function, but you *do* get a constant address which may be returned from the function, one which will continue to be valid after function was finished. This is not needed.